### ISSUE TITLE
[url_launcher] for web should support the tel and sms scheme in PWA

### ISSUE THREAD

**Issue by AngusPei** [flutter/flutter#58553]
## Use case

The current plugin supports http,https,mailto

```dart
const _mailtoScheme = 'mailto';

static final _supportedSchemes = {'http', 'https', _mailtoScheme};
```

The safari browser in the iPhone can be executed normally using the following code
```dart
  _launchURL() async {
    var url = 'tel:+1-408-555-5555';
    //if (await canLaunch(url)) {
      await launch(url);
    //} else {
      //throw 'Could not launch $url';
    //}
  }
```

## Proposal

Add support for the tel and sms scheme links in url_launcher_web, For use in pwa.

**Comment by ditman** [comment]
I think this is a very good first contribution. @AngusPei, would you be interested in contributing a fix for the plugin?

There's some special handling needed for non http/https URLs (like mailto) in PWA, so all you'd need to do is refactor `_mailtoScheme` to a Set of `_specialSchemes` that need that handling.

**Comment by twisstosin** [comment]
platform_detect doesn't seem to have an implementation to detect iOS and Android for web and it doesn't seem like it's going to be added anytime soon, there might be a need to fork and add it, https://github.com/Workiva/platform_detect/issues/41 @ditman @Vanethos

**Comment by ditman** [comment]
@twisstosin why did you need to detect iOS vs Android? Just to enable the support when on mobile?

**Comment by twisstosin** [comment]
yeah @ditman, is there a possibility of tel and sms working for desktop?

**Comment by ditman** [comment]
@twisstosin, I'm not sure if there's a built-in tel/sms handler for desktop, but for example Skype used to handle tel links:
https://mycyberuniverse.com/create-html-link-that-starts-skype-call.html

(Not sure if this is true anymore). It is impossible to detect from the browser if there's any protocol handler registered, so you need to let the user click on it at least.

**Comment by ditman** [comment]
This feature is now published, on `url_launcher_web 0.1.2`
